packages/apollo-federation/src/composition/validate/postComposition/keySelectionSetsDeclared.ts

```diff
@@ -0,0 +1,91 @@
+import { GraphQLSchema, isObjectType, FieldNode, GraphQLError } from 'graphql';
+
+import { logServiceAndType, errorWithCode } from '../../utils';
+
+/**
+ * - Each key's selection set must be present on the base service
+ */
+export const keySelectionSetsDeclared = (schema: GraphQLSchema) => {
+  const errors: GraphQLError[] = [];
+
+  const types = schema.getTypeMap();
+  for (const [typeName, namedType] of Object.entries(types)) {
+    if (!isObjectType(namedType)) continue;
+
+    if (namedType.federation && namedType.federation.keys) {
+      const baseService = namedType.federation.serviceName;
+      if (!baseService) {
+        // Federated type does not have a base service. This shouldn't happen, but does in some tests.
```

Comment:
Now that I understand the case, can we update the comment to say something like: "This is an invalid case which is handled by the `UniqueTypeNamesWithFields` validator, throwing error `VALUE_TYPE_NO_ENTITY`"

Reply:
Okay, done